code/game/atoms.dm

```diff
@@ -145,6 +145,11 @@
 		stack_trace("Warning: [src]([type]) initialized multiple times!")
 	flags_1 |= INITIALIZED_1
 
+	if(loc)
+		SEND_SIGNAL(loc, COMSIG_ATOM_ENTERED, src) // if the new item is created over a turf, send a signal that it entered that turf, `loc` in this case
+		for(var/atom/A in loc) // send a signal for every item on the `loc` turf to let them know a new item just crossed it
+			SEND_SIGNAL(A, COMSIG_MOVABLE_CROSSED, src)

```

Comment:
Just send on the turf, things in the turf can register on the turf if they need to know this

Reply:
I'm not sure I understand what you mean here.

Reply:
You don't need to send the signal to all the contents because anything in those contents can listen to the signal on their loc instead

Reply:
Ok so if I understand, I should ditch these two lines here, but keep the one above it for ATOM_ENTERED. Then in my case I would just register the conveyor to listen for that signal on its `loc`?

Reply:
yes